### DISCUSSION TITLE
树形递归删除问题，数据库Mariadb(相当于mysql)

### DISCUSSION THREAD

**Discussion by s899000** [dotnetcore/FreeSql#1277] [Q&A]
![image](https://user-images.githubusercontent.com/33239560/193167939-f71d4e61-cf8d-49f4-af7a-38efecb6eab0.png)
生成的sql在数据库中是执行成功的，但是代码确报错
DELETE FROM `Dic` WHERE (`Id` in (select * from (SELECT a.`Id` 
FROM ( SELECT cte_tbc.cte_level, a.`Id`, a.`Code`, a.`Pid`, a.`Text`, a.`Value`
  FROM (
    SELECT @cte_ids as cte_ids, (
      SELECT @cte_ids := group_concat(`Id`) 
          FROM `Dic` 
          WHERE find_in_set(`Pid`, @cte_ids)
    ) as cte_cids, @cte_level := @cte_idcte_levels + 1 as cte_level
    FROM `Dic`, (
      SELECT @cte_ids := a.`Id`, @cte_idcte_levels := 0 
          FROM `Dic` a
      WHERE 1=1 AND (a.`Id` = 16)
      LIMIT 1) cte_tbb
    WHERE @cte_ids IS NOT NULL
  ) cte_tbc, `Dic` a
  WHERE find_in_set(a.`Id`, cte_tbc.cte_ids) ) a 
ORDER BY a.cte_level DESC) ftb_del))

![image](https://user-images.githubusercontent.com/33239560/193168135-89b1a8a8-921d-4358-a550-add23a88ff28.png)

**Comment by 2881099** [comment]
使用极联删除文档，看一下

**Comment by 2881099** [comment]
https://github.com/dotnetcore/FreeSql/wiki/%E8%81%94%E7%BA%A7%E5%88%A0%E9%99%A4

**Comment by s899000** [comment]
树形，不是级联

**Comment by s899000** [comment]
就是照文档来的写的，生成的sql执行没问题，但是代码会报错

**Comment by s899000** [comment]
https://freesql.net/guide/select-as-tree.html#_1%E3%80%81totreelist

**Comment by luoyunchong** [comment]
连接串增加 Allow User Variables=True

**Reply by s899000** [reply]
OK了

**Comment by 2881099** [comment]
树形是可以极联删除的。

你用的递归查询可能 in 内部使用了参数变量，百度搜索一下错误关键字。
